fog/ledger/server/src/db_fetcher.rs

```diff
@@ -133,25 +139,51 @@ impl<DB: Ledger, E: LedgerEnclaveProxy + Clone + Send + Sync + 'static> DbFetche
                 log::info!(self.logger, "Db fetcher thread stop requested.");
                 break;
             }
+
+            // Hack: If we notice that we are way behind the ledger, set ourselves unready
+            match self.db.num_blocks() {
+                Ok(num_blocks) => {
+                    if num_blocks > self.next_block_index + 100 {
+                        self.readiness_indicator.set_unready();
+                    }
+                }
+                Err(err) => {
+                    log::error!(self.logger, "Could not get num blocks from db: {}", err);
+                }
+            };
+
             // Each call to load_block_data attempts to load one block for each known
             // invocation. We want to keep loading blocks as long as we have data to load,
             // but that could take some time which is why the loop is also gated
             // on the stop trigger in case a stop is requested during loading.
             while self.load_block_data() && !self.stop_requested.load(Ordering::SeqCst) {}
```

Comment:
I wonder if there should be a check in here to potentially unready. It seems like it's possible that more data is added during this loop that would indicate that we should unready ourselves?

Reply:
yeah that seems possible

the unready thing is kind of a hack but i think its correct that loading into ORAM is slower than downloading data. So maybe the unready check should be in this loop